dotnet/src/webdriver/Interactions/Actions.cs

```diff
@@ -147,35 +112,36 @@ public Actions SetActiveKeyboard(string name)
         /// </summary>
         /// <param name="name">The name of the wheel device to set as active.</param>
         /// <returns>A self-reference to this Actions class.</returns>
+        /// <exception cref="InvalidOperationException">If a device with this name exists but is not a wheel.</exception>
         public Actions SetActiveWheel(string name)
         {
-            IList<ActionSequence> sequences = this.actionBuilder.ToActionSequenceList();
+            InputDevice device = FindDeviceById(name);
+
+            this.activeWheel = device switch
+            {
+                null => new WheelInputDevice(name),
+                WheelInputDevice wheelDevice => wheelDevice,
+                _ => throw new InvalidOperationException($"Device under the name \"{name}\" is not a wheel. Actual input type: {device.DeviceKind}"),
+            };
 
-            InputDevice device = null;
+            return this;
+        }
 
-            foreach (var sequence in sequences)
+        private InputDevice FindDeviceById(string name)
```

Comment:
Even it is private, would be beneficial to make it `bool TryFindDeviceById(..., out ...)` given that it might return `null`? Or it is good example to use NRT here?

Reply:
I really wanted to make this more DRY, but the best error messages + assignment to specific `this.activeX` make this hard to do.

If this were more than a couple of lines which are used only 3 times, then maybe it would be worth it to have more abstraction. But as it is, this is probably the simplest way.

> it is good example to use NRT here?

The only really good NRT improvements we can make is adding `[MemberNotNull(nameof(activeWheel))]` to `SetActiveWheel`, which will tell the compiler that `this.activeWheel` does not need a null check after that method has been called. But that's waiting for the nullability attributes in the other PR.

Reply:
If NRT is our case, then let's do it later as separate initiative.

Reply:
I agree, no need to make this PR do too much

Reply:
All right.